src/mappings/core.ts

```diff
@@ -351,6 +357,17 @@ export function handleMint(event: Mint): void {
 
   updateTokenDayData(token0 as Token, event)
   updateTokenDayData(token1 as Token, event)
+
+  // update unique day entity
+  let uniqueDailyInteractionsData = updateUniqueDailyInteractions(event)
+
+  // increment unique daily swaps
+  if (!uniqueDailyInteractionsData.addresses.includes(mint.to as Bytes)) {
```

Comment:
This condition creates a race condition between `swap`, `burn`, and `mint` events. 

Whichever the user does first on any given day is recorded. In this handler, we record only the first `mint`s for this user. Subsequent events **for this user** are not recorded. As stated in the problem we just want the unique wallets that interacted with Swapr on that day, we can just store the address list in `DailyUniqueInteraction`. Having no data is better than inaccurate data.

Reply:
> What do you mean? The handleMint handler is triggered on every Mint event right?

I should have elaborated on this with some examples: 

Address `0x001` goes to Swapr and adds liquidity to SWPR-WETH pool on Arbitrum. Then, `DailyUniqueInteraction` looks something like this 
  ```
{
     "data": {
       "dailyUniqueInteractions": {
          "addresses": ["0x001"],
          "mints": [
             { to: "0x001" }, // 
          ],
          "swaps": [], // empty; no swaps on that day yet
          "burns": [] // empty; no burns on that day yet
  }```
  
Then,  Address `0x001` comes to Swap page, and swaps some DAI to WETH. After that, `DailyUniqueInteraction` looks something like this 
  ```
{
     "data": {
       "dailyUniqueInteractions": {
          "addresses": ["0x001"], // nothing changed 
          "mints": [
             { to: "0x001" }, // 
          ],
          "swaps": [], // empty! Something is wrong  
          "burns": [] // empty!
  }```

As seen, the `swaps` is empty because 
```
!uniqueDailyInteractionsData.addresses.includes("0x001") 
````

will be always true until next day at 00:00 AM UTC. And that explains the race condition that leads to missing actions for "0x001" for that day.